src/dotnet/commands/dotnet-compile/Program.cs

```diff
@@ -71,16 +71,35 @@ private static bool OnExecute(List<ProjectContext> contexts, CompilerCommandApp
 
             var managedOutput = outputPaths.CompilationFiles.Assembly;
 
+            // Create the library exporter
+            var exporter = context.CreateExporter(args.ConfigValue);
```

Comment:
Do you mean the rid? It comes from `compile` option for both cross and native compilation. If unspecified, defaults to platform rid. I see error handling missing here for other specified rids unknown to platform ILC.

Otherwise, coreclr and corert target dotnet core and should be compatible for the most part except few assemblies which we will acquire through the aot rid or put in the appdep which gets the first preference.